src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -145,12 +147,31 @@ private static void CodeGenerateRazorComponents(GeneratorExecutionContext contex
             for (var i = 0; i < files.Count; i++)
             {
                 var (hint, sourceText) = outputs[i];
-                context.AddSource(hint, sourceText);
+                context.AddSource(hint, ResolveSourceTextFromCache(sourceText, hint));
```

Comment:
Could we cache this based on the input rather than the output so we basically no-op the codegeneration i.e.

```C#
var file = files[i]; // Line 129
if (TryGetFromSource(file.AdditionalText.Path out var cachedSourceText) && file.AdditionalText.GetChecksum().Equals(cachedSourceText.GetCheck())
{
    outputs[i] = (hint, sourceText);
}
else
{
   // Do work
  // Update cache
}
```

Also it might be interesting to see if we could share some of this code between views and razor since the two code paths are almost identical

Reply:
Refactored this up a bit. Let me know what you think.